fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/JobsServlet.java

```diff
@@ -164,8 +165,15 @@ public void getJobs(
                                     )
                     );
 
-            apiRequest.getJobViews()
-                    .toList()
+            Observable<Map<String, String>> jobViews;
+
+            if (filters == null || "".equals(filters)) {
+                jobViews = apiRequest.getJobViews();
+            } else {
+                jobViews = apiRequest.getFilteredJobViews(apiRequest.buildJobStoreFilter(filters));
+            }
```

Comment:
I think it would be better if we did what the `DataServlet` does: pass in the filters to the `JobsApiRequest` and let it parse the filters internally. Then, when `getJobViews()` is invoked, the `JobsApiRequest` applies the filters if present. So, in the `DataServlet`, the code would look like:

```java
            JobsApiRequest apiRequest = new JobsApiRequest (
                    format,
                    null,  //asyncAfter is null so it behaves like a synchronous request
                    perPage,
                    page,
                    filters, //Note the addition of the filters to the constructor
                    uriInfo,
                    jobPayloadBuilder,
                    apiJobStore,
                    preResponseStore,
                    broadcastChannel
            );
... //Other stuff
jobViews = apiRequest.getJobViews(); //Inside JobsApiRequest::getJobViews, JobsApiRequest::getFilteredJobViews is invoked if there are filters
```